enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,255 @@
+
+One proposal for a minimal DoD was:
+- Feature gate in openshift/api
+    - API types moved to openshift/api
+    - CRD Initial dev https://github.com/openshift/api/pull/1929
+        Add controller-gen logic to makefile
+        Add API to api/config/v1
+        Add Generated CRD: config/v1/zz_generated.crd-manifests/0000_10_config-operator_01_clustermonitoring.crd.yaml
+        Add example CustomResource: config/examples/clustermonitoringoperator.yaml
+    - Client codegen 
+    - Reconcile logic: https://github.com/openshift/cluster-monitoring-operator/pull/2350
+    - Add decoupling ConfigMap / CustomResource:
+        Controller logic is strongly dependant of *manifests.Config struct.
+        
+
+### Example configuration
+
+
+#### current ConfigMap
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    telemeterClient:
+      enabled: false
+    prometheusK8s:
+      volumeClaimTemplate:
+        metadata:
+          name: prometheus-data
+          annotations:
+            openshift.io/cluster-monitoring-drop-pvc: "yes"
+        spec:
+          resources:
+            requests:
+              storage: 20Gi
+```
+
+### CRD
+
+```
+apiVersion: clustermonitoring.config.openshift.io
+kind: ClusterMonitoring
+metadata:
+  name: cluster
+  namespace: openshift-monitoring
+spec:
+  prometheusK8s:
+    volumeClaimTemplate:
+      metadata:
+        name: prometheus-data
+        annotations:
+          openshift.io/cluster-monitoring-drop-pvc: "yes"
+      spec:
+        resources:
+          requests:
+            storage: 20G
+```
+
+
+### Workflow Description
+
+See example in the [Proposal](#Proposal) for details.
+
+### API Extensions
+
+Add two new CRD's:
+- ClusterMonitoring
+- UserWorkloadClusterMonitoring
+
+
+### Migration path
+
+
+- Switch mechanism
+- Both ConfigMap and CRD will coexist for a while until the transition is complete.
+- CRD and ConfigMap will be merged into a single structure, this structure will config CMO
+- Extensive testing will be required to validate the new behavior.
+- If there are different fields in CRD and ConfigMap, the field in the ConfigMap will take precedence.
+- Some fields in CRD and ConfigMap diverge due to API requirements. Helper functions will be needed to resolve this.
+- The end user will be able to choose to use ConfigMap or CRD at any time, keeping in mind that the fields in the CRD will gradually expand.
+
+### Risks and Mitigations
+
+- Issues when ConfigMap and CR coexist
+- Mismatch in the configuration of operators with ConfigMap and CR
+
+### Drawbacks
+
+- Migration to user
+- Coexisting of both ConfigMap and CRD
+
+
+### Test Plan
+
+- Unit tests for the feature
+- e2e tests covering the feature
+- Upgrade testing
+
```

Comment:
I'd also put an item for the FeatureGate

Reply:
done